internal/logger/io_sink.go

```diff
@@ -82,14 +212,22 @@ func (osSink *IOSink) Info(_ int, msg string, keysAndValues ...interface{}) {
 		kvMap[keysAndValues[i].(string)] = keysAndValues[i+1]
 	}
 
-	switch msg {
-	case CommandStarted:
-		logCommandMessageStarted(osSink.log, kvMap)
-	case CommandSucceeded:
-		logCommandMessageSucceeded(osSink.log, kvMap)
-	case CommandFailed:
-		logCommandMessageFailed(osSink.log, kvMap)
-	}
+	map[string]func(*log.Logger, map[string]interface{}){
+		CommandStarted:            logCommandStartedMessage,
+		CommandSucceeded:          logCommandSucceededMessage,
+		CommandFailed:             logCommandFailedMessage,
```

Comment:
A much more memory-efficient approach is to define this command -> function mapping as a package variable.

E.g.
```go
var commandFunc = map[string]func(*log.Logger, map[string]interface{}){
	CommandStarted:            logCommandStartedMessage,
	CommandSucceeded:          logCommandSucceededMessage,
	CommandFailed:             logCommandFailedMessage,
	// ...
}

func (osSink *IOSink) Info(...) {
	f := commandFunc[msg]
	if f != nil {
		f(osSink.log, kvMap)
		return
	}

	// Appropriate logger not found, use the default message formatter.
	logDefault(osSink.log, kvMap)
}
```